source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -235,16 +250,20 @@ def filter_too_short_heavy_journeys(journey, request):
             continue
 
         if s.street_network.mode == response_pb2.Car \
-                and exceed_min_duration(s, journey, request, request['_min_car']):
+                and min_car is not None \
+                and exceed_min_duration(s, journey, min_duration=min_car,
+                                        orig_modes=orig_modes, dest_modes=dest_modes):
             logger.debug("the journey {} has not enough car, we delete it".format(journey.internal_id))
             mark_as_dead(journey, is_debug, "not_enough_car")
-            return False or is_debug
+            return is_debug
```

Comment:
IMHO, i just find this less understandable

Reply:
TBH, it's still obscure to me...
Why would we want make the caller know about `is_debug` ?! As long as we mark a journey dead, what difference does it make ? I'm confused.

Reply:
@nonifier FYI, there'll be a better version for this

Reply:
```python
    def wrapper(filter, is_debug, filter_msg):
        def wrapped_filter(journey):
            res = filter(journey)
            if not res:
                mark_as_dead(journey, is_debug, filter_msg)            
            return res or is_debug
        return wrapped_filter
```
use this at this line https://github.com/CanalTP/navitia/pull/2438/files#diff-43ca03fe9a32e6f5f976232379c763cd